cmd/cluster-ingress-operator/main.go

```diff
@@ -38,34 +43,75 @@ func main() {
 	if err != nil {
 		logrus.Fatalf("Failed to get watch namespace: %v", err)
 	}
-	kubeClient := k8sclient.GetKubeClient()
 
-	ic, err := util.GetInstallConfig(kubeClient)
+	handler, err := createHandler(namespace)
 	if err != nil {
-		logrus.Fatalf("could't get installconfig: %v", err)
-	}
-
-	cvoClient, err := cvoclientset.NewForConfig(k8sclient.GetKubeConfig())
-	if err != nil {
-		logrus.Fatalf("Failed to get cvoClient: %v", err)
-	}
-
-	handler := &stub.Handler{
-		CvoClient:       cvoClient,
-		InstallConfig:   ic,
-		Namespace:       namespace,
-		ManifestFactory: manifests.NewFactory(),
+		logrus.Fatalf("couldn't create handler: %v", err)
 	}
 
 	if err := handler.EnsureDefaultClusterIngress(); err != nil {
 		logrus.Fatalf("failed to ensure default cluster ingress: %v", err)
 	}
+
 	resyncPeriod := 10 * time.Minute
 	logrus.Infof("Watching %s, %s, %s, %d", resource, kind, namespace, resyncPeriod)
 	sdk.Watch(resource, kind, namespace, resyncPeriod)
 	// TODO Use a named constant for the router's namespace or get the
 	// namespace from config.
 	sdk.Watch("apps/v1", "DaemonSet", "openshift-ingress", resyncPeriod)
+	sdk.Watch("v1", "Service", "openshift-ingress", resyncPeriod)
 	sdk.Handle(handler)
 	sdk.Run(context.TODO())
 }
+
+func createHandler(namespace string) (*stub.Handler, error) {
+	cvoClient, err := cvoclientset.NewForConfig(k8sclient.GetKubeConfig())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create CVO client: %v", err)
+	}
+
+	ic, err := util.GetInstallConfig(k8sclient.GetKubeClient())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get installconfig: %v", err)
+	}
+
+	var dnsManager dns.Manager
+	switch {
+	case ic.Platform.AWS != nil:
```

Comment:
We could have an `InstallConfig` with a nil `Platform`, so we should check that here.

Reply:
Ah, right, sorry.  Never mind!